src/App.jsx

```diff
@@ -1,38 +1,147 @@
 import logo from "/logo.png";
+
 import "./App.css";
-import { makeShuffledDeck } from "./utils.jsx";
+
 import { useState } from "react";
 
+import { makeShuffledDeck } from "./utils.jsx";
+import UI from "./UI.jsx";
+import Card from "./Card.jsx";
+
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+
 function App(props) {
   // Set default value of card deck to new shuffled deck
   const [cardDeck, setCardDeck] = useState(makeShuffledDeck());
   // currCards holds the cards from the current round
   const [currCards, setCurrCards] = useState([]);
+  // condition to check if game
+  const [inGame, setInGame] = useState(true);
+  const [inRound, setInRound] = useState(true);
+  const [endMessage, setEndMessage] = useState("");
+  const [score1, setScore1] = useState(0);
+  const [score2, setScore2] = useState(0);
 
   const dealCards = () => {
     const newCurrCards = [cardDeck.pop(), cardDeck.pop()];
     setCurrCards(newCurrCards);
+    setInRound(true);
   };
   // You can write JavaScript here, just don't try and set your state!
 
-  // You can access your current components state here, as indicated below
-  const currCardElems = currCards.map(({ name, suit }) => (
-    // Give each list element a unique key
-    <div key={`${name}${suit}`}>
-      {name} of {suit}
-    </div>
-  ));
+  const checkCards = (cards) => {
+    if(cards.length != 0)
+    {
+      if(cards[0].rank > cards[1].rank)
+      {
+        setScore1(score1 + 1);
+        setInRound(false);
+        return (
+          <p>Player 1 won this round!</p>
+        );
+      }
+      else if(cards[1].rank > cards[0].rank)
+      {
+        setScore2(score2 + 1);
+        setInRound(false);
+        return (
+          <p>Player 2 won this round!</p>
+        );
+      }
+      else if(cards[0].rank == cards[1].rank)
+      {
+        if(cards[0].suit == 'Spades')
+        {
+          setScore1(score1 + 1);
+          setInRound(false);
+          return (
+            <p>Player 1 won this round!</p>
+          );
+        }
+        else if(cards[0].suit == 'Hearts' && cards[1].suit == "Clubs" || cards[1].suit == "Diamonds")
+        {
+          setScore1(score1 + 1);
+          setInRound(false);
+          return (
+            <p>Player 1 won this round!</p>
+          );
+        }
+        else if(cards[0].suit == 'Clubs' && cards[1].suit == "Diamond")
+        {
+          setScore1(score1 + 1);
+          setInRound(false);
+          return (
+            <p>Player 1 won this round!</p>
+          );
+        }
```

Comment:
I think cards could be assigned a `value` property or something similar based on the suit. This could reduce this tie code by a lot

Reply:
Also, these 3 if conditions here, all do the same thing. We could just reduce that to a single if condition.